packages/flutter/lib/src/material/animated_raster.dart

```diff
@@ -0,0 +1,202 @@
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, covariant _RenderAnimatedRaster renderObject) {
+    renderObject
+      ..animation = animation
+      ..delegate = delegate
+      ..devicePixelRatio = MediaQuery.maybeOf(context)?.devicePixelRatio ?? 1.0;
+  }
+}
+
+/// A render object that converts its children to a [ui.Image].
+///
+/// This render object will create a GPU resident texture the first time that
+/// [_AnimatedRasterDelegate.willPaint] returns true and and provide it to
+/// [_AnimatedRasterDelegate.paintRaster]. This will be reused until `willPaint`
+/// returns false. It is up to the implementor of the raster delegate to handle
+/// painting both the rasterized and the non-rasterized child.
+///
+/// See also:
+///  * [ZoomPageTransitionsBuilder], which uses this render object to implement the fade and
+///    stretch effect efficiently.
+class _RenderAnimatedRaster extends RenderProxyBox {
+  /// Create a new [_RenderAnimatedRaster].
+  _RenderAnimatedRaster(
+    this._animation,
+    this._delegate,
+    this._devicePixelRatio,
+  );
+
+  /// The delegate that controls how the child raster is painted.
+  _AnimatedRasterDelegate get delegate => _delegate;
+  _AnimatedRasterDelegate _delegate;
+  set delegate(_AnimatedRasterDelegate value) {
+    if (value == delegate) {
+      return;
+    }
+    _delegate = value;
+    markNeedsPaint();
+  }
+
+  /// The device pixel ratio used to create the child raster.
+  double get devicePixelRatio => _devicePixelRatio;
+  double _devicePixelRatio;
+  set devicePixelRatio(double value) {
+    if (value == devicePixelRatio) {
+      return;
+    }
+    _devicePixelRatio = value;
+    markNeedsPaint();
+  }
+
+  /// The animation used to drive this render object.
+  Animation<double> get animation => _animation;
+  Animation<double> _animation;
+  set animation(Animation<double> value) {
+    if (value == animation) {
+      return;
+    }
+    animation.removeListener(markNeedsPaint);
+    _animation = value;
+    animation.addListener(markNeedsPaint);
+    markNeedsPaint();
+  }
+
+  @override
+  void dispose() {
+    animation.removeListener(markNeedsPaint);
+    _childImage?.dispose();
+    _childImage = null;
+    super.dispose();
+  }
+
+  @override
+  void attach(covariant PipelineOwner owner) {
+    animation.addListener(markNeedsPaint);
+    super.attach(owner);
+  }
+
+  @override
+  void detach() {
+    animation.removeListener(markNeedsPaint);
+    super.detach();
+  }
+
+  // Paint [child] with this painting context, then convert to a raster and detach all
+  // children from this layer.
+  ui.Image _paintAndDetachToGpuImage() {
+    final OffsetLayer offsetLayer = layer! as OffsetLayer;
+    final PaintingContext context = PaintingContext(offsetLayer, Offset.zero & size);
+    context.paintChild(child!, Offset.zero);
+    // ignore: invalid_use_of_protected_member
```

Comment:
Ugh...

Reply:
Yeah, this is why I added the PaintingContext API